ibm/service/kms/resource_ibm_kms_key.go

```diff
@@ -166,103 +166,24 @@ func ResourceIBMKmskey() *schema.Resource {
 }
 
 func resourceIBMKmsKeyCreate(d *schema.ResourceData, meta interface{}) error {
-	instanceID := getInstanceIDFromCRN(d.Get("instance_id").(string))
-	kpAPI, _, err := populateKPClient(d, meta, instanceID)
+	kpAPI, keyData, err := ExtractAndValidateKeyDataFromSchema(d, meta)
 	if err != nil {
 		return err
 	}
 
-	kpAPI.Config.KeyRing = d.Get("key_ring_id").(string)
-
-	name := d.Get("key_name").(string)
-	standardKey := d.Get("standard_key").(bool)
-
-	var expiration *time.Time
-	if es, ok := d.GetOk("expiration_date"); ok {
-		expiration_string := es.(string)
-		// parse string to required time format
-		expiration_time, err := time.Parse(time.RFC3339, expiration_string)
-		if err != nil {
-			return fmt.Errorf("[ERROR] Invalid time format (the date format follows RFC 3339): %s", err)
-		}
-		expiration = &expiration_time
-	} else {
-		expiration = nil
+	key, err := kpAPI.CreateImportedKey(context.Background(), keyData.Name, keyData.Expiration, keyData.Payload, keyData.EncryptedNonce, keyData.IV, keyData.Extractable)
```

Comment:
Since we are collapsing different condition checks and pushing the logic to Go SDK, you need to add test case evidences satisfying all type of scenarios